auction-app-frontend/src/components/Searchbar/Searchbar.jsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { search } from "src/assets/icons";
+
+import "./style.scss";
+
+const Searchbar = () => {
+  const [query, setQuery] = useState("");
+
+  const navigate = useNavigate();
+
+  const handleSearch = () => {
+    const url = new URL(window.location.href);
+    url.searchParams.set("search", query);
+
+    if (location.pathname.includes('/shop/')) {
+      navigate(url.pathname + url.search);
+    } else {
+      navigate(`/shop/?search=${query}`);
+    }
+  };
+
+  return (
+    <div className="search-bar">
+        <input
+        type="text"
+        className="search-input"
+        placeholder="Try enter: Shoes"
+        value={ query }
+        onChange={ (event) => setQuery(event.target.value) }
```

Comment:
Check this
you can create method, that accepts event as parameter and write this 
onChange={method} 

It will do the same, but we will avoid issue with creating method with every render. 
Please apply this in other places also.